core/blockchain.go

```diff
@@ -2066,25 +2065,33 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 		if err != nil {
 			return it.index, err
 		}
-		// /Quorum
+		// End Quorum
 
 		// If we have a followup block, run that against the current state to pre-cache
 		// transactions and probabilistically some of the account/storage trie nodes.
 		var followupInterrupt uint32
 		if !bc.cacheConfig.TrieCleanNoPrefetch {
 			if followup, err := it.peek(); followup != nil && err == nil {
 				throwaway, _ := state.New(parent.Root, bc.stateCache, bc.snaps)
-				//due to privateStateRepo changes, for now only do prefetch on the public state
-				//we do have an option to also do it on overall privateState but need a function in privateStateRepo to return the overall private stateCache
-				//this reverts the prefetch definition to the original go-ethereum definition (only takes in publicState)
-				go func(start time.Time, followup *types.Block, throwaway *state.StateDB, interrupt *uint32) {
-					bc.prefetcher.Prefetch(followup, throwaway, bc.vmConfig, &followupInterrupt)
-
-					blockPrefetchExecuteTimer.Update(time.Since(start))
-					if atomic.LoadUint32(interrupt) == 1 {
-						blockPrefetchInterruptMeter.Mark(1)
-					}
-				}(time.Now(), followup, throwaway, &followupInterrupt)
+
+				// Quorum
+				privateStateRepo, stateRepoErr := bc.privateStateManager.StateRepository(parent.Root)
+				if privateStateRepo != nil {
+					throwawayPrivateStateRepo := privateStateRepo.Copy()
+
+					// Quorum: add privateStateThrowaway argument
+					go func(start time.Time, followup *types.Block, throwaway *state.StateDB, privateStateThrowaway mps.PrivateStateRepository, interrupt *uint32) {
+						bc.prefetcher.Prefetch(followup, throwaway, throwawayPrivateStateRepo, bc.vmConfig, &followupInterrupt)
+
+						blockPrefetchExecuteTimer.Update(time.Since(start))
+						if atomic.LoadUint32(interrupt) == 1 {
+							blockPrefetchInterruptMeter.Mark(1)
+						}
+					}(time.Now(), followup, throwaway, throwawayPrivateStateRepo, &followupInterrupt)
+				} else if stateRepoErr != nil {
+					log.Warn("Unable to load the private state repository for pre-fetching", "stateRepoErr", stateRepoErr)
+				}
```

Comment:
I would put the standard go lang error handling, I think there is no case where err and private state repo are not nil same time.

if err != nil {
  log.Warn("Unable to load the private state repository for pre-fetching", "stateRepoErr", stateRepoErr)
} else {
...
}
but here is small change to handle the error with minimal change:
```suggestion
				if err == nil && privateStateRepo != nil {
					throwawayPrivateStateRepo := privateStateRepo.Copy()

					// Quorum: add privateStateThrowaway argument
					go func(start time.Time, followup *types.Block, throwaway *state.StateDB, privateStateThrowaway mps.PrivateStateRepository, interrupt *uint32) {
						bc.prefetcher.Prefetch(followup, throwaway, throwawayPrivateStateRepo, bc.vmConfig, &followupInterrupt)

						blockPrefetchExecuteTimer.Update(time.Since(start))
						if atomic.LoadUint32(interrupt) == 1 {
							blockPrefetchInterruptMeter.Mark(1)
						}
					}(time.Now(), followup, throwaway, throwawayPrivateStateRepo, &followupInterrupt)
				} else {
					log.Warn("Unable to load the private state repository for pre-fetching", "stateRepoErr", stateRepoErr)
				}
```